### DISCUSSION TITLE
When creating a table through parquet file and then querying, some statements report error --- IO Error: Could not read all bytes from files

### DISCUSSION THREAD

**Discussion by ethink** [duckdb/duckdb#3521] [General]
Hello
I create a table through parquet file, and then some statements report errors when querying. The statement to create a table is as follows：
 	        

	        public static void createTableFromFile() {

	        try {
				Class.forName("org.duckdb.DuckDBDriver");
			
	        Properties ro_prop = new Properties();
			ro_prop.setProperty("duckdb.read_only", "false");
			 Connection conn = DriverManager.getConnection("jdbc:duckdb:C:\\test\\duckdb\\test743.db",ro_prop);

	        Statement stmt = conn.createStatement();
	      
	        stmt.execute("CREATE TABLE people743w AS SELECT * FROM parquet_scan('E:\\rdh\\duckdb\\d\\taxi_2019_04.parquet');");
	       
	
	        stmt.execute("CREATE TABLE people2193w AS SELECT * FROM parquet_scan('E:\\rdh\\duckdb\\d\\alltaxi.parquet');");
	       stmt.execute("CREATE TABLE people1yi AS SELECT * FROM parquet_scan('E:\\rdh\\duckdb\\d\\a1yi.parquet');");
	       
	        stmt.close();
	        conn.close();
	        } catch (Exception e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
	    
		}

The query code is as follows：
		
		

		
		public static void main(String[] args) {
		System.out.println("++++++++++++++");
		String sql1="SELECT *  FROM people743w limit 2";
		queryTest(sql1);
		
		String sql2="SELECT pickup_at,sum(fare_amount) FROM people743w group by pickup_at limit 1000";
		queryTest(sql2);
	}
 		public static int queryTest(String sql) {
		  int count=0;
			    System.out.println("Start query statement:"+sql);
			    try
			    {
		//			      Properties ro_prop = new Properties();
		//					ro_prop.setProperty("duckdb.read_only", "false");
					DuckDBConnection conn2 = (DuckDBConnection) 						DriverManager.getConnection("jdbc:duckdb:C:\\test\\duckdb\\test743.db");
					Statement stmt = conn2.createStatement();
					ResultSet rs = stmt.executeQuery(sql);
					rs.close(); 
			      stmt.close();
			      conn2.close();
			      System.out.println("query was successful!");
			    }catch(Exception e) {
			    	e.printStackTrace();
			    }
		  
		  return count;
	  }

The error log is as follows：
++++++++++++++
Start query statement:SELECT *  FROM people743w limit 2
query was successful!
Start query statement:SELECT pickup_at,sum(fare_amount) FROM people743w group by pickup_at limit 1000
java.sql.SQLException: IO Error: Cannot open file "C:\test\duckdb\test743.db": wanted=262144 read=0
	at org.duckdb.DuckDBNative.duckdb_jdbc_startup(Native Method)
	at org.duckdb.DuckDBDatabase.<init>(DuckDBDatabase.java:22)
	at org.duckdb.DuckDBDriver.connect(DuckDBDriver.java:35)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at TestDuckdb.queryTest(TestDuckdb.java:932)
	at TestDuckdb.main(TestDuckdb.java:62)

***Supplementary note:** alltaxi Parquet contains 100 million pieces of data. Before using this file, **"select * from people743w limit 2"**, **"select pickup_at, sum (fare_amount) from people743w group by pickup_at limit 1000"** can be executed normally D

Yes, a**fter it is created in the table, the statement "select * from people743w limit 2" can be executed, but the statement "select pickup_at, sum (fare_amount) from people743w group by pickup_at limit 1000**" will report an error if it is not executed

**Comment by thinkORo** [comment]
Good morning ething,

is there any reason that you use two different connection strings?

1. Connection conn = DriverManager.getConnection("jdbc:duckdb:C:\\test\\duckdb\\test743.db",ro_prop);
2.  DuckDBConnection conn2 = (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:C:\test\duckdb\test743.db");

The reported error " java.sql.SQLException: IO Error: Cannot open file "C:\test\duckdb\test743.db": wanted=262144 read=0" suggests that your program cannot find the file. Either it is not there (at the exact position) or it is open by any other process.

I'm not a java developer, therefore it could be that I'm wrong.

**Reply by thinkORo** [reply]
Hmm, as I can see in my own comment: the double backslashes are not shown (in the first connection string) in this comment. Therefore it could be that your code is right but the content of the issue isn't shown as expected.

Hint: I can see the double \\\ if I would edit my own comment.

My suggestion is: you are running Windows and therefore you need double backslashes for your connection string. Could please double-check?
